### PR TITLE
feat: add expect.closeTo() API

### PR DESCRIPTION
**Pull request by bartoszgolebiowski**
https://github.com/vitest-dev/vitest/issues/2809

### CODE REVIEW

File: test/core/test/jest-expect.test.ts [hunk 1]
```diff
@@ -133,7 +133,13 @@ describe('jest-expect', () => {
 
     expect('Mohammad').toEqual(expect.stringMatching(/Moh/))
     expect('Mohammad').not.toEqual(expect.stringMatching(/jack/))
-
+    expect({
+      title: '0.1 + 0.2',
+      sum: 0.1 + 0.2,
+    }).toEqual({
+      title: '0.1 + 0.2',
+      sum: expect.closeTo(0.3, 5),
```
Comment (sheremet-va) [inline test/core/test/jest-expect.test.ts:141]: Not enough tests. What about `not`? What about failing cases?

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: test/core/test/jest-expect.test.ts (hunk 1 above)
Comment (bartoszgolebiowski) [reply test/core/test/jest-expect.test.ts:141]: Sure.

**Comment by bartoszgolebiowski** [comment]
can you retrigger pipelines? @sheremet-va

File: test/core/test/jest-expect.test.ts [hunk 2]
```diff
@@ -133,6 +133,29 @@ describe('jest-expect', () => {
...
+      sum: expect.closeTo(0.4, 5),
+    })
+
+    expect({
+      title: '0.1 + 0.2',
+      sum: 0.1 + 0.2,
+    }).toEqual({
+      title: '0.1 + 0.2',
+      sum: expect.not.closeTo(0.4, 5),
+    })
```
Comment (sheremet-va) [inline test/core/test/jest-expect.test.ts:158]: Can we have a test that will actually fail?

```ts
expect(() => ...).toThrow(...)
```

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Documentation is missing. Can you add it, please? Other than that good job 👍🏻

**Comment by bartoszgolebiowski** [comment]
I will!

**Comment by skhaz** [comment]
This would be a great addition and would help with the adoption of Vitest. Thank you.

**Comment by ghiscoding** [comment]
@sheremet-va @Dunqing any chance this could be picked up by one of you and completed? It doesn't look like the original author has time to complete this PR and Jest already has this feature as [`expect.closeTo(number, numDigits)`](https://jestjs.io/docs/expect#expectclosetonumber-numdigits)

**Comment by sheremet-va** [comment]
> @sheremet-va @Dunqing any chance this could be picked up by one of you and completed? It doesn't look like the original author has time to complete this PR and Jest already has this feature as [`expect.closeTo(number, numDigits)`](https://jestjs.io/docs/expect#expectclosetonumber-numdigits)

Feel free to open a separate PR 😄

**Comment by ghiscoding** [comment]
I knew you were going to say that but I don't feel comfortable to take this one over

**Comment by Dunqing** [comment]
I will take this one over

**Pull request closed without merging**